store/neo4j.go

```diff
@@ -469,6 +471,73 @@ func (n *NeoDataStore) GetCode(ctx context.Context, codeListID, edition string,
 	return codeModel, nil
 }
 
+func (n *NeoDataStore) GetCodeDatasets(ctx context.Context, codeListID, edition, code string) (*models.Datasets, error) {
+	editionExists, err := n.EditionExists(ctx, codeListID, edition)
+	if err != nil {
+		return nil, err
+	}
+	if !editionExists {
+		return nil, datastore.ErrEditionNotFound
+	}
+
+	conn, err := n.pool.OpenPool()
+	if err != nil {
+		return nil, errors.WithMessage(err, "GetCodeDatasets: error opening neo4j connection")
+	}
+
+	defer conn.Close()
+
+	query := fmt.Sprintf(getCodeDatasets, codeListID, edition, code)
+
+	rows, err := conn.QueryNeo(query, nil)
+	if err != nil {
+		return nil, errors.WithMessage(err, "getCode: conn.QueryNeo returned an error")
+	}
+	defer rows.Close()
+
+	datasetsModel := &models.Datasets{}
+
+	err = extractRowResults(ctx, rows, func(row []interface{}, rowIndex int) error {
+		node := row[0].(graph.Node)
+		relationship := row[1].(graph.Relationship)
+
+		vars := node.Properties
+		relVars := relationship.Properties
+
+		if isPublished, ok := vars["is_published"].(bool); !ok || !isPublished {
```

Comment:
Consider filtering published datasets in the Cypher query `AND (d.is_published=true)`

```
MATCH (d)<-[inDataset]-(c:_code)-[r:usedBy]->(cl:_code_list:`_code_list_cpih1dim1aggid`) WHERE (cl.edition="one-off") AND (c.value="cpih1dim1S90401") AND (d.is_published=true) RETURN d,r
```